linetools/spectra/io.py

```diff
@@ -252,7 +256,12 @@ def readspec(specfil, inflg=None, efil=None, verbose=False, flux_tags=None,
             uwave = u.Quantity(wave, unit=u.AA)
         else:
             uwave = u.Quantity(wave)
+        # Sort
+        srt = np.argsort(uwave)
+        uwave = uwave[srt]
+        fx = fx[srt]
         if sig is not None:
+            sig = sig[srt]
             xspec1d = XSpectrum1D.from_array(uwave, u.Quantity(fx),
```

Comment:
Yes, I wish.  Unfortunately there are too many ways
to instantiate and I can't do anything about the WCS ones..